### PR TITLE
Feat/add swagger schemas documentation

### PR DESCRIPTION
**Pull request by robertotorres-dev**
This pr is to add swagger doc and its configuration.
Another feature is the schemas for each user endpoint

### CODE REVIEW

File: packages/api-gateway/src/drivers/http/plugins/services.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
-const PluginLoader = require('fastify-plugin');
+const fp = require('fastify-plugin');
```
Comment (hecttoressau) [inline packages/api-gateway/src/drivers/http/plugins/services.js:1]: Let this name var like PluginLoader, is more expressive

File: packages/api-gateway/src/drivers/http/plugins/swagger.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+const fp = require('fastify-plugin');
```
Comment (hecttoressau) [inline packages/api-gateway/src/drivers/http/plugins/swagger.js:1]: Rename this variable to something more expresive

File: packages/api-gateway/src/drivers/http/routes/usuarios/index.js [hunk 3]
```diff
@@ -1,31 +1,33 @@
 const { usuariosAdapter } = require('../../adapters');
+
 const {
+  getUsuariosSchema,
```
Comment (hecttoressau) [inline packages/api-gateway/src/drivers/http/routes/usuarios/index.js:4]: Rename this to getAllUsuariosSchema, getGroupUsuariosSchema, getListUsuariosSchema or other  is easy to mistype getUsuarioSchema and getUsuariosSchema

File: packages/api-gateway/src/drivers/http/routes/usuarios/schema/generalSchema.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+// General
+const general = {
+  createdAt: { type: 'string', format: 'date-time' },
+  updatedAt: { type: 'string', format: 'date-time' },
+  deletedAt: { type: 'string', format: 'date-time' },
+};
+
+module.exports = {
+  usuario, persona, domicilio, estado, municipio, general,
+};
```
Comment (hecttoressau) [inline packages/api-gateway/src/drivers/http/routes/usuarios/schema/generalSchema.js:67]: By the moment this is okay, but if in the future we are going to add more schemas here, we should separate them in files

File: packages/api-gateway/src/drivers/http/routes/usuarios/schema/updateUsuarioSchema.js [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+          type: 'object',
+          properties: {
+            id: usuario.id,
+            rolId: usuario.rolId,
+            usuario: usuario.usuario,
+            correo: usuario.correo,
+            estatus: usuario.estatus,
+            createdAt: general.createdAt,
+            updatedAt: general.updatedAt,
+            persona: {
```
Comment (hecttoressau) [inline packages/api-gateway/src/drivers/http/routes/usuarios/schema/updateUsuarioSchema.js:59]: We should create persona schema in other file
```
            persona:  personaSchema
```

File: packages/api-gateway/src/drivers/http/routes/usuarios/schema/getUsuariosSchema.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+            type: 'object',
+            properties: {
+              id: usuario.id,
+              rolId: usuario.rolId,
+              usuario: usuario.usuario,
+              correo: usuario.correo,
+              estatus: usuario.estatus,
+              createdAt: general.createdAt,
+              updatedAt: general.updatedAt,
+              persona: {
```
Comment (hecttoressau) [inline packages/api-gateway/src/drivers/http/routes/usuarios/schema/getUsuariosSchema.js:22]: Create persona schema in external file

File: packages/api-gateway/src/drivers/http/routes/usuarios/schema/getUsuarioSchema.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+          type: 'object',
+          properties: {
+            id: usuario.id,
+            rolId: usuario.rolId,
+            usuario: usuario.usuario,
+            correo: usuario.correo,
+            estatus: usuario.estatus,
+            createdAt: general.createdAt,
+            updatedAt: general.updatedAt,
+            persona: {
```
Comment (hecttoressau) [inline packages/api-gateway/src/drivers/http/routes/usuarios/schema/getUsuarioSchema.js:28]: Create persona schema in external file

File: packages/api-gateway/src/drivers/http/routes/usuarios/schema/getUsuarioDetalleSchema.js [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+                      type: 'object',
+                      properties: {
+                        id: estado.id,
+                        paisId: estado.paisId,
+                        estado: estado.estado,
+                        createdAt: general.createdAt,
+                        updatedAt: general.updatedAt,
+                      },
+                    },
+                    municipio: {
```
Comment (hecttoressau) [inline packages/api-gateway/src/drivers/http/routes/usuarios/schema/getUsuarioDetalleSchema.js:78]: create municipio schema in external file

File: packages/api-gateway/src/drivers/http/routes/usuarios/schema/getUsuarioDetalleSchema.js [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+                nacionalidad: persona.nacionalidad,
+                telefono: persona.telefono,
+                celular: persona.celular,
+                curp: persona.curp,
+                rfc: persona.rfc,
+                ine: persona.ine,
+                fotografia: persona.fotografia,
+                createdAt: general.createdAt,
+                updatedAt: general.updatedAt,
+                domicilio: {
```
Comment (hecttoressau) [inline packages/api-gateway/src/drivers/http/routes/usuarios/schema/getUsuarioDetalleSchema.js:53]: Create domicilio schema in external file

File: packages/api-gateway/src/drivers/http/routes/usuarios/schema/getUsuarioDetalleSchema.js [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+          properties: {
+            id: usuario.id,
+            rolId: usuario.rolId,
+            usuario: usuario.usuario,
+            correo: usuario.correo,
+            estatus: usuario.estatus,
+            createdAt: general.createdAt,
+            updatedAt: general.updatedAt,
+            persona: {
+              type: 'object',
```
Comment (hecttoressau) [inline packages/api-gateway/src/drivers/http/routes/usuarios/schema/getUsuarioDetalleSchema.js:36]: Create persona schema in external file

File: packages/api-gateway/src/drivers/http/routes/usuarios/schema/deleteUsuarioSchema.js [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+          properties: {
+            id: usuario.id,
+            rolId: usuario.rolId,
+            usuario: usuario.usuario,
+            correo: usuario.correo,
+            estatus: usuario.estatus,
+            createdAt: general.createdAt,
+            updatedAt: general.updatedAt,
+            deletedAt: general.deletedAt,
+            persona: {
```
Comment (hecttoressau) [inline packages/api-gateway/src/drivers/http/routes/usuarios/schema/deleteUsuarioSchema.js:32]: create schema persona in external file

File: packages/api-gateway/src/drivers/http/routes/usuarios/schema/createUsuarioSchema.js [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+          type: 'object',
+          properties: {
+            id: usuario.id,
+            rolId: usuario.rolId,
+            usuario: usuario.usuario,
+            correo: usuario.correo,
+            estatus: usuario.estatus,
+            createdAt: general.createdAt,
+            updatedAt: general.updatedAt,
+            persona: {
```
Comment (hecttoressau) [inline packages/api-gateway/src/drivers/http/routes/usuarios/schema/createUsuarioSchema.js:54]: create persona schema in external file

File: packages/api-gateway/src/drivers/http/routes/usuarios/schema/createUsuarioSchema.js [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+  body: {
+    type: 'object',
+    properties: {
+      rolId: usuario.rolId,
+      usuario: usuario.usuario,
+      correo: usuario.correo,
+      contrasena: usuario.contrasena,
+      estatus: usuario.estatus,
+      actualizado: usuario.actualizado,
+      persona: {
```
Comment (hecttoressau) [inline packages/api-gateway/src/drivers/http/routes/usuarios/schema/createUsuarioSchema.js:19]: create persona schema files in external file

File: packages/api-gateway/src/drivers/http/routes/usuarios/schema/domicilioResponseSchema.js [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+    calle: domicilio.calle,
+    numeroExterior: domicilio.numeroExterior,
+    numeroInterior: domicilio.numeroInterior,
+    colonia: domicilio.colonia,
+    codigoPostal: domicilio.codigoPostal,
+    latitud: domicilio.latitud,
+    longitud: domicilio.longitud,
+    createdAt: general.createdAt,
+    updatedAt: general.updatedAt,
+    estado: {
```
Comment (hecttoressau) [inline packages/api-gateway/src/drivers/http/routes/usuarios/schema/domicilioResponseSchema.js:23]: Create estado schema in external file

File: packages/api-gateway/src/drivers/http/routes/usuarios/schema/createUsuarioSchema.js (hunk 13 above)
Comment (hecttoressau) [reply packages/api-gateway/src/drivers/http/routes/usuarios/schema/createUsuarioSchema.js:17]: You already have persona schema in other file, use it here

File: packages/api-gateway/src/drivers/http/routes/usuarios/schema/domicilioResponseSchema.js [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+      type: 'object',
+      properties: {
+        id: estado.id,
+        estado: estado.estado,
+        createdAt: general.createdAt,
+        updatedAt: general.updatedAt,
+      },
+    },
+    municipio: {
```
Comment (hecttoressau) [inline packages/api-gateway/src/drivers/http/routes/usuarios/schema/domicilioResponseSchema.js:32]: Create municipio schema in external file

File: packages/api-gateway/src/drivers/http/routes/usuarios/schema/updateUsuarioSchema.js [hunk 16]
```diff
@@ -0,0 +1,86 @@
...
+  },
+  body: {
+    type: 'object',
+    properties: {
+      rolId: usuario.rolId,
+      usuario: usuario.usuario,
+      correo: usuario.correo,
+      estatus: usuario.estatus,
+      actualizado: usuario.actualizado,
+      persona: {
```
Comment (hecttoressau) [inline packages/api-gateway/src/drivers/http/routes/usuarios/schema/updateUsuarioSchema.js:26]: You already have a file for persona schema use it here

File: packages/api-gateway/src/drivers/http/routes/usuarios/schema/updateUsuarioSchema.js (hunk 5 above)
Comment (hecttoressau) [reply packages/api-gateway/src/drivers/http/routes/usuarios/schema/updateUsuarioSchema.js:59]: You already have persona schema in external file use it here

File: packages/api-gateway/src/drivers/http/routes/usuarios/schema/updateUsuarioSchema.js (hunk 16 above)
Comment (robertotorres-dev) [reply packages/api-gateway/src/drivers/http/routes/usuarios/schema/updateUsuarioSchema.js:26]: It's a little different the request schema than the response schema. How can I merge that schema?

**Review by hecttoressau — APPROVED** [review]
LGTM

**Pull request merged**